rfcs/0107-version-normalization.md

```diff
@@ -0,0 +1,206 @@
+       the mentioned unlabeled snapshot was distributed.
+
+- Some atypical programs have special considerations:
+  - Linux kernel modules:
+    - Besides the rules established above for typical snapshots (whether labeled
+      or unlabeled), `version` shoud have appended `+linux=XX.XX.XX`, where
+      `XX.XX.XX` is the corresponding Linux kernel version used to build the
+      aforementioned module.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Some useful examples:
+
+- Bochs is a typical Sourceforge-hosted project; its labeled snapshots can be
+  fetched from tarballs obtained via URLs like
+  <https://sourceforge.net/projects/bochs/files/bochs/2.6.11/>
+
+  For this example, we have `pname = "bochs"; version = "2.6.11";`.
+
+- MPV is a typical Github-hosted program; its labeled snapshots can be fetched
+  from tarballs obtained via URLs like
+  <https://github.com/mpv-player/mpv/releases/tag/v0.33.1>.
+
+  For this example, we get rid of the `"v"` prepended to the version tag: `pname
+  = "mpv"; version = "0.33.1";`.
+
+- SDL2 is hosted on Github; its latest labeled version can be downloaded from
+  <https://github.com/libsdl-org/SDL/releases/tag/release-2.0.14>. Therefore we
+  have `pname = "SDL2"; version = "2.0.14";`.
+
+  _However_, this labeled version was released December 21, 2020, while the
+  latest change was done in May 28, 2021.
+
+  Therefore, for this particular unlabeled releases of SDL2, we have `pname =
+  "SDL2"; version = "2.0.14+unstable=2021-05-28";`.
+
+- Cardboard is a typical Gitlab-hosted program. It has no labeled release yet,
+  therefore we use `0.pre` as default dummy stable version; further, the latest
+  commit was made on May 10, 2021.
+
+  Therefore, we have `pname = "cardboard"; version =
+  "0.pre+unstable=2021-05-21";`.
+
+- Python is a famous programming language and interpreter. Before the
+  deprecation of its 2.x series in 2020, Python had two release branches,
+  popularly known as 'Python 2' and 'Python 3'. Indeed this peculiar situation
+  reflected in many package managers, especially Nixpkgs, that employed
+  `python2` and `python3` as `pname`s for these particular programs.
+
+  As an exercise of imagination, suppose the scenarios described below:
+
+  Python 2.6 was released 2008-10-01; an unlabeled snapshot of Python 2 branch
+  released at 2008-12-04 would have `version="2.6+unstable=2008-12-04"`.
+
+  At the same time, Python 3.0 was released 2008-12-03; an unlabeled snapshot of
+  Python 3 branch released at 2008-12-04 would have
+  `version="3.0+unstable=2008-12-04"`.
+
+- The Linux drivers for Realtek rtl8192eu can be fetched from a Github page,
+  <https://github.com/Mange/rtl8192eu-linux-driver>. It has no labeled release;
+  the latest code is from May 12, 2021. Supposing e.g. it was built for Linux
+  kernel version 5.10.01, we therefore have `pname = "rtl8192eu"; version =
+  "0.pre+unstable=2021-05-12+linux=5.10.01";`.
+
+# Drawbacks
```

Comment:
Repology compatibility and formats used by other distributions needs investigation here.

If we break compatibility with the version formats present on Repology, we create more noise for maintainers using this as a tool and may even impede some automation which uses repology as a data source.

Reply:
I talk about this with the [Repology maintainer](https://github.com/repology/repology-updater/issues/345). He thinks [it is a good format](https://github.com/repology/repology-updater/issues/345#issuecomment-915269874), however he will not implement it because it is not the prevalent format in Nixpkgs.